File: 2024/May/25/CPP/main.cpp
#include <iostream>
#include <string>
#include "Tree.h"

int main()
{
    Tree<string> *treeB = new Tree<string>("B");
    Tree<string> *treeA = new Tree<string>("A");
    Tree<string> *treeD = new Tree<string>("D");
    Tree<string> *treeC = new Tree<string>("C");
    Tree<string> *treeE = new Tree<string>("E");
    Tree<string> *treeF = new Tree<string>("F");
    Tree<string> *treeG = new Tree<string>("G");
    treeB->addChild(treeA);
    treeB->addChild(treeD);
    treeD->addChild(treeC);
    treeD->addChild(treeE);
    treeB->addChild(treeF);
    treeE->addChild(treeG);

    cout << "Total de filhos de B: " << treeB->childrenCount() << endl;
    for (int f = 0; f < treeB->childrenCount(); f++)
        cout << treeB->getChild(f)->item << endl;    

    cout << "Pré-ordem:" << endl;
    vector<Tree<string> *> preordem = treeB->preOrder();
    for (int i = 0; i < preordem.size(); i++)
        cout << "-> " << preordem[i]->item << endl;

    cout << endl << "Pós-ordem:" << endl;
    vector<Tree<string> *> posordem = treeB->postOrder();
    for (int i = 0; i < posordem.size(); i++)
        cout << "-> " << posordem[i]->item << endl;
    cout << endl;
    cout << "Número de nós: " << treeB->size() << endl;

    cout << "Nível do B: " << treeB->level() << endl;
    cout << "Nível do C: " << treeC->level() << endl;
    cout << "Nível do F: " << treeF->level() << endl;

    cout << endl;
    cout << "Altura da árvore: " << treeB->height() << endl;

    delete treeA;
    delete treeB;
    delete treeC;
    delete treeD;
    delete treeE;
    delete treeF;
    delete treeG;
}

File: 2024/May/25/Main.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

/**
 *
 * @author flash
 */
public class Main {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        Tree<String> treeB = new Tree<>("B");
        Tree<String> treeA = new Tree<>("A");
        Tree<String> treeD = new Tree<>("D");
        Tree<String> treeC = new Tree<>("C");
        Tree<String> treeE = new Tree<>("E");
        Tree<String> treeF = new Tree<>("F");
        Tree<String> treeG = new Tree<>("G");
        treeB.addChild(treeA);
        treeB.addChild(treeD);
        treeB.addChild(treeF);
        treeD.addChild(treeC);
        treeD.addChild(treeE);
        treeE.addChild(treeG);

        System.out.println("Total de filhos de B: " + treeB.childrenCount());
        for (int f = 0; f < treeB.childrenCount(); f++)
            System.out.println(treeB.getChild(f).item);        
        System.out.println();

        System.out.println("Pré-ordem:");
        List<Tree<String>> preordem = treeB.preOrder();
        for (Tree<String> no : preordem)
            System.out.println("-> " + no.item);

        System.out.println("Pós-ordem:");
        List<Tree<String>> posOrdem = treeB.postOrder();
        for (Tree<String> no : posOrdem)
            System.out.println("-> " + no.item);
        
        System.out.println();
        System.out.println("Número de nós: "+treeB.size());

        System.out.println();
        System.out.println("Nível do B: "+treeB.level());
        System.out.println("Nível do C: "+treeC.level());
        System.out.println("Nível do F: "+treeF.level());

        System.out.println();
        System.out.println("Altura da árvore: "+treeB.height());
    }
}
